pkg/controller/manager/nad/controller.go

```diff
@@ -96,6 +96,11 @@ func (h Handler) OnChange(key string, nad *cniv1.NetworkAttachmentDefinition) (*
 		return nil, nil
 	}
 
+	if nad.Spec.Config == "" || nad.Labels == nil || nad.Labels[utils.KeyVlanLabel] == "" ||
```

Comment:
suggest to encap the nad default/empty check into one function, and reuse in `OnRemove`